src/test/helper.ts

```diff
@@ -5,38 +5,124 @@
+    assert.deepStrictEqual(actualItem.kind, expectedItem.kind);
+  });
 }
 
-export const testFolderPath = path.resolve(__dirname, '..', '..', 'testFixture');
+export async function testHover(docUri: vscode.Uri, position: vscode.Position, expectedCompletionList: vscode.Hover[]) {
+  const actualhover = (await vscode.commands.executeCommand(
+    'vscode.executeHoverProvider',
+    docUri,
+    position,
+  )) as vscode.Hover[];
 
-export const getDocPath = (p: string): string => {
-  return path.resolve(__dirname, '../../testFixture', p);
-};
-export const getDocUri = (p: string): vscode.Uri => {
-  return vscode.Uri.file(getDocPath(p));
-};
+  assert.equal(actualhover.length, expectedCompletionList.length);
+  expectedCompletionList.forEach((expectedItem, i) => {
+    const actualItem = actualhover[i];
+    assert.deepStrictEqual(actualItem.contents, expectedItem.contents);
+  });
+}
+
+export async function testDefinitions(
+  docUri: vscode.Uri,
+  position: vscode.Position,
+  expectedDefinitions: vscode.Location[],
+) {
+  const actualDefinitions = await vscode.commands.executeCommand<vscode.Location[] | vscode.LocationLink[]>(
+    'vscode.executeDefinitionProvider',
+    docUri,
+    position,
+  );
+
+  assert.equal(actualDefinitions.length, expectedDefinitions.length);
+  expectedDefinitions.forEach((expectedItem, i) => {
+    const actualItem = actualDefinitions[i];
+    if (actualItem instanceof vscode.Location) {
+      assert.deepStrictEqual(actualItem.uri.path, expectedItem.uri.path);
+      assert.deepStrictEqual(actualItem.range.start, expectedItem.range.start);
+      assert.deepStrictEqual(actualItem.range.end, expectedItem.range.end);
+      return;
+    } else {
+      // } else if (actualItem instanceof vscode.LocationLink) {
+      assert.deepStrictEqual(actualItem.targetUri.path, expectedItem.uri.path);
+      assert.deepStrictEqual(actualItem.targetRange.start, expectedItem.range.start);
+      assert.deepStrictEqual(actualItem.targetRange.end, expectedItem.range.end);
+    }
+  });
+}
+
+export async function testReferences(
+  docUri: vscode.Uri,
+  position: vscode.Position,
+  expectedDefinitions: vscode.Location[],
+) {
+  const actualDefinitions = await vscode.commands.executeCommand<vscode.Location[]>(
+    'vscode.executeReferenceProvider',
+    docUri,
+    position,
+  );
+
+  assert.equal(actualDefinitions.length, expectedDefinitions.length);
+  expectedDefinitions.forEach((expectedItem, i) => {
+    const actualItem = actualDefinitions[i];
+    assert.deepStrictEqual(actualItem.uri.path, expectedItem.uri.path);
+    assert.deepStrictEqual(actualItem.range.start, expectedItem.range.start);
+    assert.deepStrictEqual(actualItem.range.end, expectedItem.range.end);
+  });
+}
+
+export async function testSymbols(docUri: vscode.Uri, symbolNames: string[]) {
+  const symbols = (await vscode.commands.executeCommand(
+    'vscode.executeDocumentSymbolProvider',
+    docUri,
+  )) as vscode.SymbolInformation[];
```

Comment:
```suggestion
  const symbols = await vscode.commands.executeCommand<vscode.SymbolInformation[]>(
    'vscode.executeDocumentSymbolProvider',
    docUri,
  );
```

We can pass the type argument to `executeCommand` here, like you did for the other helpers.